Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/Mandate/Notification.hs

```diff
@@ -79,7 +81,15 @@ sendPDNNotificationToDriver Job {id, jobInfo} = withLogTag ("JobId-" <> id.getId
             setIsNotificationSchedulerRunningKey startTime endTime merchantOpCityId serviceName False
             driverFeesPostRetries <- QDF.findDriverFeeInRangeWithNotifcationNotSentServiceNameAndStatus merchantId merchantOpCityId limit startTime endTime retryCount DF.PAYMENT_PENDING serviceName
             mapM_ handleNotificationFailureAfterRetiresEnd (driverFeesPostRetries <&> (.id))
-            scheduleJobs transporterConfig startTime endTime merchantId merchantOpCityId maxShards serviceName
+            let jobDataT :: Text = show jobData
+            hashedJobData <- getHash jobDataT
+            Hedis.whenWithLockRedis (mkScheduleJobLockKey hashedJobData "Notification") 60 $ do
+              duplicationKey :: Maybe Bool <- Hedis.get $ jobDuplicationPreventionKey hashedJobData "Notification"
```

Comment:
can we use just one key ? setNx, it will return bool for the one which is able to set and if its true then only create, this will help you avoid race condition and have duplication with one key.

Reply:
but setNx doesn't have any expiry